src/services/EnvironmentVariables.ts

```diff
@@ -29,49 +30,130 @@ export class EnvironmentVariables {
 
 	private readonly GOV_NOTIFY_QUEUE_URL = process.env.GOV_NOTIFY_QUEUE_URL;
 
+	private readonly KMS_KEY_ARN = process.env.KMS_KEY_ARN;
+
+	private readonly CLIENT_CONFIG = process.env.CLIENT_CONFIG;
+
+	private readonly ENCRYPTION_KEY_IDS = process.env.ENCRYPTION_KEY_IDS;
+
+	private readonly AUTH_SESSION_TTL = process.env.AUTH_SESSION_TTL;
+
+	private readonly SIGNING_KEY_IDS = process.env.SIGNING_KEY_IDS;
+
+	private readonly JWKS_BUCKET_NAME = process.env.JWKS_BUCKET_NAME;
+
+	private readonly TXMA_QUEUE_URL = process.env.TXMA_QUEUE_URL;
+
+	private readonly PERSON_IDENTITY_TABLE_NAME = process.env.PERSON_IDENTITY_TABLE_NAME;
+
+	private readonly YOTICALLBACKURL = process.env.YOTICALLBACKURL;
+
 	/**
 	 * Constructor reads all necessary environment variables and stores them as class data.
 	 * It also performs validation on env variable values. If certain variables have unexpected values the constructor will throw an error and/or log an error message
 	 *
-	 * @param GOVUKNOTIFY_TEMPLATE_ID
-	 * @param GOVUKNOTIFY_MAX_RETRIES
-	 * @param GOVUKNOTIFY_BACKOFF_PERIOD_MS
-	 * @param YOTI_SDK
-	 * @param YOTIBASEURL
-	 * @param ISSUER
-	 * @param SESSION_TABLE
-	 * @param YOTI_KEY_SSM_PATH
-	 * @param GOVUKNOTIFY_API_KEY_SSM_PATH
 	 */
-	constructor(logger: Logger) {
-
-		if (!this.ISSUER || this.ISSUER.trim().length === 0 ||
-			!this.SESSION_TABLE || this.SESSION_TABLE.trim().length === 0 ||
-			!this.YOTI_KEY_SSM_PATH || this.YOTI_KEY_SSM_PATH.trim().length === 0 ||
-			!this.GOVUKNOTIFY_API_KEY_SSM_PATH || this.GOVUKNOTIFY_API_KEY_SSM_PATH.trim().length === 0) {
-			logger.error(`GovNotifyService - Misconfigured external API's key ${EnvironmentVariables.name}`);
-			throw new AppError(HttpCodesEnum.SERVER_ERROR, Constants.ENV_VAR_UNDEFINED);
-		}
-
-		if (!this.GOVUKNOTIFY_BACKOFF_PERIOD_MS
-			|| this.GOVUKNOTIFY_BACKOFF_PERIOD_MS.trim().length === 0
-			|| +this.GOVUKNOTIFY_BACKOFF_PERIOD_MS.trim() === 0
-			|| +this.GOVUKNOTIFY_BACKOFF_PERIOD_MS.trim() >= 60000) {
-			this.GOVUKNOTIFY_BACKOFF_PERIOD_MS = "20000";
-			logger.warn("GOVUKNOTIFY_BACKOFF_PERIOD_MS env var is not set. Setting to default - 20000");
-		}
-
-		if (!this.GOVUKNOTIFY_MAX_RETRIES
-			|| this.GOVUKNOTIFY_MAX_RETRIES.trim().length === 0
-			|| +this.GOVUKNOTIFY_MAX_RETRIES.trim() >= 100) {
-			this.GOVUKNOTIFY_MAX_RETRIES = "3";
-			logger.warn("GOVUKNOTIFY_MAX_RETRIES env var is not set. Setting to default - 3");
-		}
-
-		if (!this.YOTI_SDK || this.YOTI_SDK.trim().length === 0
-			|| !this.YOTIBASEURL || this.YOTIBASEURL.trim().length === 0) {
-			logger.error("Environment variable YOTI_SDK or YOTIBASEURL is not configured");
-			throw new AppError(HttpCodesEnum.SERVER_ERROR, Constants.ENV_VAR_UNDEFINED);
+	constructor(logger: Logger, serviceType: ServicesEnum) {
+		switch (serviceType) {
```

Comment:
could the variables in [Globals](https://github.com/alphagov/di-ipv-cri-f2f-api/blob/5b7697bc557124c0296dbb2878d62a1554a75c9f/deploy/template.yaml#L202) be moved out into specific lambdas which are using them pls unless its common across all the lambdas

Reply:
Moved all the variables to specific lambda functions as required and kept only the once used commonly across all functions. Tested all services post these changes and working fine.